src/elements/common/sub-header/GridViewButton.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+import React from 'react';
+import { injectIntl, FormattedMessage } from 'react-intl';
+
+import Button from '../../../components/button';
+import IconGridViewInverted from '../../../icons/general/IconGridViewInverted';
+import IconGridView from '../../../icons/general/IconGridView';
+import messages from '../messages';
+import Tooltip from '../Tooltip';
+import type { ViewMode } from '../flowTypes';
+import { VIEW_MODE_GRID, VIEW_MODE_LIST } from '../../../constants';
+import './GridViewButton.scss';
+
+type Props = {
+    onViewModeChange: Function,
+    viewMode: ViewMode,
+} & InjectIntlProvidedProps;
+
+const GridViewButton = (props: Props) => {
+    const { onViewModeChange, intl, viewMode, ...rest } = props;
+
+    const isGridView = viewMode === VIEW_MODE_GRID;
+
+    const onClick = () => {
+        if (isGridView) {
```

Comment:
can convert this to a one-liner: `onViewModeChange(isGridView ? VIEW_MODE_LIST : VIEW_MODE_GRID`